### ISSUE TITLE
Test discovery 'fails' when SPM emits a warning

### ISSUE THREAD

**Issue by Joannis** [swift-server/vscode-swift#569]
**Describe the bug**
One of my dependencies has a warning, and emitting that causes test discovery to fail.

**To Reproduce**
Create a package with async-http-client and MongoKitten 7.7 (not higher, as the warning is fixed soon)

**Expected behavior**
Tests load, despite a warning.

**Environment**
 - OS: macOS
 - Swift version: `Apple Swift version 5.9-dev (LLVM 3653146fb31d979, Swift 889f813e44d8d04)
Target: arm64-apple-macosx13.0`
 - Visual Studio Code version: 1.80.0
 - vscode-swift version: 1.4.0

**Additional context**

**Comment by adam-fowler** [comment]
Yeah if I remember right this is related to SPM outputting errors to stdout and not stderr.

**Comment by adam-fowler** [comment]
This has the added issue where it prints out the `[1/1] Planning build` text after warning me about the dependencies even though I asked it to skip the build.

It seems to fix itself after the second build though

**Comment by adam-fowler** [comment]
@Joannis do you want to try this with the latest Xcode beta. It seems to working ok now
